extensions/amp-intersection-observer-polyfill/0.1/amp-intersection-observer-polyfill.js

```diff
@@ -27,7 +28,9 @@ const TAG = 'amp-intersection-observer-polyfill';
 function upgrade(win) {
   upgradePolyfill(win, () => {
     installIntersectionObserver();
-    // TODO(dvoytenko): hookup 3p host updater for inabox.
+    if (win != win.top) {
+      maybeSetupCrossOriginObserver(win);
```

Comment:
Won't this bring the cross-origin listener into every page? Eg, it'll now install an inabox listener for regular amp pages.

Reply:
No. `maybeSetupCrossOriginObserver` takes care of this by checking we are actually running in a host-enabled AMP iframe such as inabox. The details of how that happens are kind of messy, so I wanted to keep it hidden inside the `maybeSetupCrossOriginObserver` method. This also makes it easier to test.

Reply:
Reworked to put it all in the `maybeSetupCrossOriginObserver` and added more docs there. I that's a little clearer now.